### PR TITLE
Custom generators for plain ruby classes

### PR DESCRIPTION
**Pull request by redtachyons**
1. Service object
2. Form object
3. Query object

### CODE REVIEW

File: lib/generators/form_object/USAGE [hunk 1]
```diff
@@ -0,0 +1,8 @@
+Description:
+    Creates service object
+
+Example:
+    rails generate form_object Thing
+
+    This will create:
+        app/services/thing.rb
```
Comment (coderhs) [inline lib/generators/form_object/USAGE]: If its a form object then we need it in `app/forms`

File: lib/generators/rspec/form_object/form_object_generator.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+module Rspec
+class FormObjectGenerator < Rails::Generators::NamedBase
```
Comment (coderhs) [inline lib/generators/rspec/form_object/form_object_generator.rb]: Intending.

File: lib/generators/service_object/templates/service_object.rb.tt [hunk 3]
```diff
@@ -0,0 +1,18 @@
+<% module_namespacing do -%>
+class <%= class_name %>
```
Comment (coderhs) [inline lib/generators/service_object/templates/service_object.rb.tt]: In the main template and project, in general, I would like to keep format `ClassNameService`, didn't want to break the convention in clickx but here would like to keep the postfix service.

**Review by coderhs — CHANGES_REQUESTED** [review]

**Review by coderhs — APPROVED** [review]

**Comment by coderhs** [comment]
@tachyons I have made changes to your `Custom Generator`. Basically, I changed the nomenclature used. Do share your review.

**Comment by redtachyons** [comment]
LGTM

**Pull request merged**